File: leetcode/0283-move-zeroes/solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        put = 0
        
        for pick in range(len(nums)):
            if nums[pick] != 0:
                nums[put] = nums[pick]
                put += 1
            
        while put < len(nums):
            nums[put] = 0
            put += 1

File: leetcode/0283-move-zeroes/solution.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int put = 0;
        
        for (int pick = 0; pick < nums.size(); pick++) {
            if (nums[pick] != 0) {
                nums[put] = nums[pick];
                put++;
            }
        }
        
        while (put < nums.size()) {
            nums[put] = 0;
            put++;
        }
    }
};